test/tools/unified-spec-runner/runner.ts

```diff
@@ -199,6 +199,21 @@ export async function runUnifiedTest(
           actualEvents = clientCommandEvents.get(clientId);
         }
         expect(actualEvents, `No client entity found with id ${clientId}`).to.exist;
+        const wasProvidedCredentials = !!entities.getEntity('client', clientId).options.credentials;
+
+        // TODO(NODE-2471): Currently the node driver executes a ping command when credentials are provided
+        // Here, we need to filter out two extra events from a set of CMAP events.
+        if (
+          wasProvidedCredentials &&
+          actualEvents.length > 4 &&
```

Comment:
why specifically > 4? even with the expected events being at index 1 & 2, don't we need just `>= 3` to apply this logic?

Reply:
I added this condition before also checking for credentials I may be able to drop it. But it seemed like there were tests (failure cases) that expected to see connectionPoolReady and connectionCheckOut, but not followed by a checkIn. In that case even though it is the ping command being the operation that caused the failure, it still proves the failure occurs. (I'll get back to you on the test name).

Reply:
I'm not following, will be interested to see what fails and why

Reply:
whoops [added back in a check to make sure the array has at least one item in it](https://spruce.mongodb.com/version/617074eb562343589588413e/tasks)